### PR TITLE
Quick start for existing users V2 - Changes to tour completion

### LINKED ISSUE #16352: Quick Start for Existing Users V2 - Changes to Tour Completion
Parent #16347

- ~~Replace blue circle with a green circle and tick on tour completion~~ (deferred to take into account new design)
- [x] Show snackbar with more guidance link on tour completion @AjeshRPai

### PR DESCRIPTION
**Pull request by AjeshRPai**
Parent #16352 

Description
Added a Congrats notice that's displayed when all tours are complete

Final Screenshot 
Screenshot_20220510_153902

---
### Pre requisites 

Since the plans feature is removed from the site menu and not from the quick start(https://github.com/wordpress-mobile/WordPress-Android/issues/16477), there is no way to complete the Explore Plans task at present. In order to complete the quick start tasks, we will have to skip the Explore plans feature from the quick start tour. 

We can do that by going to the quick start prompt and long-pressing on Explore plan and selecting skip tasks.  The below video capture shows the steps to skip the Plans tasks as well as the Completed tour notice being displayed. 

Video Capture : Time stamp
0:03 - Skipping Explore Plans Task 
0:15-0:20 - Completed tour notice being shown

https://user-images.githubusercontent.com/17463767/167681712-3cc9812e-cc21-4efc-897b-a7183353b12f.mp4

### Known Issue 
[Quick Start showing Plans item despite app not showing plan information](https://github.com/wordpress-mobile/WordPress-Android/issues/16477)

To test
- Login to a Site with quick start in Progress
- Go through all the tours
- Verify: once you're back on the My Site screen, the congrats notice is displayed
- Verify: The congrats notice is automatically dismissed after 5 seconds (or you can manually dismiss it by tapping on it)

**Continue and Verify that Completed Notice is not displayed again** 
- Switch to the Reader tab
- Switch back to the My Site tab
- Verify: The congrats notice isn't displayed
- Force quit the app
- Open the app
- Verify: The congrats notice isn't displayed

## Regression Notes
1. Potential unintended areas of impact
Completed Notice being shown incorrectly 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit tests and manual tests 

3. What automated tests I added (or what prevented me from doing so)
Unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by osullivanchris — CHANGES_REQUESTED** [review]
@AjeshRPai Hey I installed this build but I haven't been able to complete the tour. For every site it keeps asking me to Explore Plans. But then there is no Plans option in the list to tap. Its happening both for a self hosted site and a dotcom site.

Otherwise the attached video looks good with 1 other issue: the text is flipped in the video. It should be

**Congrats! You know your way around.** Doesn't it feel good to cross things off a list?

At the moment in your video the two sentences are flipped. They look correct in the image you put at the top.

**Comment by AjeshRPai** [comment]
@osullivanchris Thanks for testing the changes, As you pointed out the issues 

>For every site it keeps asking me to Explore Plans. But then there is no Plans option in the list to tap. Its happening both for a self hosted site and a dotcom site

This is a known issue, the only workaround, for now, is to skip the Explore plans task and complete the quick start tour. The attached video shows how to do that. I have also linked the Explore plans as a known issue in the PR description 

>Otherwise the attached video looks good with 1 other issue: the text is flipped in the video. It should be

It was an Incorrect Video, I have replaced it with the correct one

**Comment by leandroalonso** [comment]
> For every site it keeps asking me to Explore Plans. But then there is no Plans option in the list to tap. Its happening both for a self hosted site and a dotcom site

@AjeshRPai @ashiagr I know this is not related to our changes in this project, but can't we simply remove the "Explore Plans" task while Plans isn't visible?

**Comment by ashiagr** [comment]
👋 @leandroalonso 

Actually, we're thinking about whether we should remove the plan quick start task or update `/plans/mobile` endpoint to return correct plans and re-enable plans. As mentioned here (p1651688458935559/1651587009.445519-slack-C011BKNU1V5), removing a quick start task might not be straightforward. 

@AjeshRPai can you time-box and check how much effort it can take?

**Comment by AjeshRPai** [comment]
@leandroalonso 

I believe an effort of 3-4 days is needed for removing the plans entirely and since it’s temporary, we will need to make some effort while re-introducing the entire flow with new pricing. So I believe reintroducing the plans item on site menu and showing the URL(https://wordpress.com/pricing/) in a Webview when clicking on it is better for now. 

In future, we can take up the task of updating the endpoint  `/plans/mobile` at a later stage once we have the bandwidth. Let me know your thoughts. 

cc: @ashiagr

**Review by osullivanchris — APPROVED** [review]
> This is a known issue, the only workaround, for now, is to skip the Explore plans task and complete the quick start tour. The attached video shows how to do that. I have also linked the Explore plans as a known issue in the PR description

Got it!

> It was an Incorrect Video, I have replaced it with the correct one

Looks better now! Approving but could you also just double check the padding/line spacing is the same on that new snackbar. The line-spacing looks a little bit tight. And the top and bottom padding around the text looks a bit large. But if its inheriting the same values as the other ones we can leave it for now.

File: WordPress/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -3131,6 +3131,10 @@
     <string name="quick_start_focus_point_description">tap here</string>
     <string name="quick_start_site_menu_tab_message_short">Tap &lt;b&gt;%1$s&lt;/b&gt; to continue.</string>
 
+    <string name="quick_start_completed_tour_title">Congrats! You know your way around&lt;br/&gt;</string>
+    <string name="quick_start_completed_tour_subtitle">Doesn\'t it feel good to cross things off a list?</string>
+
+
```
Comment (ashiagr) [inline WordPress/src/main/res/values/strings.xml]: Nitpick: There are double line spaces, one empty line can be removed.

**Review by ashiagr — APPROVED** [review]
Good job @AjeshRPai! Changes look good to me. I just left one nitpick. 

Also, a reminder that [quick start dynamic cards show a category completion message](https://github.com/wordpress-mobile/WordPress-Android/blob/b8fe0e2c7bce524c79c99aa994bfb8318f646d59/WordPress/src/main/java/org/wordpress/android/ui/mysite/cards/quickstart/QuickStartRepository.kt#L225-L231) which might coincide with this all tasks completion message. I couldn't test it as I couldn't find skip task optiion for dynamic card -> Plans task. Since it is behind a feature flag, let's not worry about it, but we can create an issue with existing GH label `My Site Dynamic Cards` to track it.

Feel free to merge the PR when you're ready.

**Comment by AjeshRPai** [comment]
@osullivanchris 

>The line-spacing looks a little bit tight. And the top and bottom padding around the text looks a bit large. But if its inheriting the same values as the other ones we can leave it for now.

I have used the same component and haven't done any changes related to the UI behaviour of that component(snackbar), It's the same.

File: WordPress/src/main/res/values/strings.xml (hunk 1 above)
Comment (AjeshRPai) [reply WordPress/src/main/res/values/strings.xml]: @ashiagr  👍🏼  Removed in [6a4fc13](https://github.com/wordpress-mobile/WordPress-Android/pull/16505/commits/6a4fc1351f90766dcdf2d48d32b43c73378f08e8)

**Pull request merged**